mpas_analysis/shared/plot/plotting.py

```diff
@@ -1,75 +1,125 @@
 
-    titleFontSize is the size of the title font
+    titleFontSize : int, optional
+        the size of the title font
 
-    figsize is the size of the figure in inches
+    figsize : tuple of float, optional
+        the size of the figure in inches
 
-    dpi is the number of dots per inch of the figure
+    dpi : int, optional
+        the number of dots per inch of the figure
 
-    Author: Xylar Asay-Davis
+    maxXTicks : int, optional
+        the maximum number of tick marks that will be allowed along the x axis.
+        This may need to be adjusted depending on the figure size and aspect
+        ratio.
 
-    Last Modified: 02/11/2017
+    Authors
+    -------
+    Xylar Asay-Davis
+
+    Last Modified
+    -------------
+    03/14/2017
     """
-    fig = plt.figure(figsize=figsize, dpi=dpi)
+    plt.figure(figsize=figsize, dpi=dpi)
 
+    minDays = []
+    maxDays = []
     for dsIndex in range(len(dsvalues)):
         dsvalue = dsvalues[dsIndex]
         if dsvalue is None:
             continue
         mean = pd.Series.rolling(dsvalue.to_pandas(), N, center=True).mean()
         mean = xr.DataArray.from_series(mean)
-        plt.plot_date(mean['Time'], mean,
-                      fmt=lineStyles[dsIndex],
-                      xdate=True,
-                      ydate=False,
-                      linewidth=lineWidths[dsIndex])
+        minDays.append(mean.Time.min())
+        maxDays.append(mean.Time.max())
+        plt.plot(mean['Time'], mean,
+                 lineStyles[dsIndex],
+                 linewidth=lineWidths[dsIndex])
+
+    start = days_to_datetime(np.amin(minDays), calendar=calendar)
+    end = days_to_datetime(np.amax(maxDays), calendar=calendar)
+
+    if (end.year - start.year > maxXTicks/2):
+        major = [date_to_days(year=year, calendar=calendar)
+                 for year in np.arange(start.year, end.year+1)]
+        formatterFun = partial(_date_tick, calendar=calendar,
+                               includeMonth=False)
+    else:
+        # add ticks for months
+        major = []
+        for year in range(start.year, end.year+1):
+            for month in range(1, 13):
+                major.append(date_to_days(year=year, month=month,
+                                          calendar=calendar))
+        formatterFun = partial(_date_tick, calendar=calendar,
+                               includeMonth=True)
 
     ax = plt.gca()
-    # TODO: it would be good to change labels to include months if the plot is
-    # shorter than a couple of years
-    ax.xaxis.set_major_locator(mdates.YearLocator())
-    ax.xaxis.set_major_formatter(mdates.DateFormatter('%Y'))
-    ax.xaxis.set_minor_locator(mdates.MonthLocator())
-    fig.autofmt_xdate()
+    ax.xaxis.set_major_locator(FixedLocator(major, maxXTicks))
+    ax.xaxis.set_major_formatter(FuncFormatter(formatterFun))
+
+    plt.setp(ax.get_xticklabels(), rotation=30)
+
+    plt.autoscale(enable=True, axis='x', tight=True)
```

Comment:
@milenaveneziani, here's the command to make the x axis "tight" in case you need it elsewhere.